sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -1192,8 +1192,8 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
   }
 
   test("SPARK-10740: handle nondeterministic expressions correctly for set operations") {
-    val df1 = (1 to 20).map(Tuple1.apply).toDF("i")
-    val df2 = (1 to 10).map(Tuple1.apply).toDF("i")
+    val df1 = spark.range(1, 20).select('id.cast("int").as("i"))
+    val df2 = spark.range(1, 10).select('id.cast("int").as("i"))
```

Comment:
This syntax `(1 to 20).map(Tuple1.apply).toDF("i")` produces a `LocalRelation`. These `LocalRelation`s are subsequently `Union`'ed. The new optimizer rules reduces this Union into a single LocalRelation. Which fails this test, because the new approach results in an already evaluated `LocalRelation` (using a different seed for the RNG), instead of a `Union` with two separate partitions.